frontend/src/contexts/EntriesContext.tsx

```diff
@@ -0,0 +1,60 @@
+import {
+  createContext,
+  useContext,
+  useRef,
+  useState,
+  useMemo,
+  useCallback,
+  ReactNode,
+} from 'react';
+import type { PatchTypeWithKey } from '@/hooks/useConversationHistory';
+
+interface EntriesContextType {
+  entries: PatchTypeWithKey[];
+  setEntries: (entries: PatchTypeWithKey[]) => void;
+  reset: () => void;
+}
+
+const EntriesContext = createContext<EntriesContextType | null>(null);
+
+interface EntriesProviderProps {
+  children: ReactNode;
+}
+
+export const EntriesProvider = ({ children }: EntriesProviderProps) => {
+  const entriesRef = useRef<PatchTypeWithKey[]>([]);
+  const [, forceRerender] = useState(0);
```

Comment:
This doesn't seem to do anything as consumers of `value` won’t see changes since it doesn't depend on `_`.